fenix/app/src/main/java/org/mozilla/fenix/gleanplumb/MessageNotificationWorker.kt

```diff
@@ -159,8 +160,13 @@ class MessageNotificationWorker(
             val instanceWorkManager = WorkManager.getInstance(context)
             instanceWorkManager.enqueueUniquePeriodicWork(
                 MESSAGE_WORK_NAME,
-                // We want to keep any existing scheduled work.
-                ExistingPeriodicWorkPolicy.KEEP,
+                // We want to keep any existing scheduled work, unless
+                // when we're under test.
+                if (messaging.isUnderTest()) {
```

Comment:
I think ideally we would not have logic for testing purposes. Wonder if we modify HomeActivityIntentTestRule and pass in the ExistingPeriodicWorkPolicy to setMessageNotificationWorker